CharlieBackend.Business/Services/AccountService.cs

```diff
@@ -158,6 +145,39 @@ public async Task<bool> DisableAccountAsync(long id)
             }
         }
 
+        public async Task<Result<AccountDto>> ChangePasswordAsync(ChangeCurrentPasswordDto changePasswd)
+        {
+            var user = await _unitOfWork.AccountRepository.GetAccountCredentialsByEmailAsync(changePasswd.Email);
+            
+            if (user != null)
+            {
+                var salt = await _unitOfWork.AccountRepository.GetAccountSaltByEmail(changePasswd.Email);
+
+                if (salt != "")
+                {
+                    string checkPassword = HashPassword(changePasswd.CurrentPassword, salt);
+
+                    if (user.Password == checkPassword)
+                    {
+                        user.Salt = GenerateSalt();
+                        user.Password = HashPassword(changePasswd.NewPassword, user.Salt);
+
+                        _unitOfWork.AccountRepository.UpdateAccountCredentials(user);
```

Comment:
You don't need to call this method at all (You've already updated salt and password). Actually, it can be removed